internal/store/somestore/metrics.go

```diff
@@ -0,0 +1,104 @@
+package somestore
+
+import (
+	"context"
+	"sync"
+
+	"github.com/VadimOcLock/metrics-service/internal/entity/enum"
+	"github.com/VadimOcLock/metrics-service/internal/errorz"
+)
+
+type MemStorage struct {
+	mu       sync.RWMutex
+	gauges   map[string]float64
+	counters map[string]int64
+}
+
+type UpdateGaugeMetricParams struct {
+	Name  string
+	Value float64
+}
+
+func (i *Impl) UpdateGaugeMetric(ctx context.Context, arg UpdateGaugeMetricParams) (bool, error) {
+	i.s.mu.Lock()
+	defer i.s.mu.Unlock()
+	i.s.gauges[arg.Name] = arg.Value
+
+	return true, nil
+}
+
+type UpdateCounterMetricParams struct {
+	Name  string
+	Value int64
+}
+
+func (i *Impl) UpdateCounterMetric(ctx context.Context, arg UpdateCounterMetricParams) (bool, error) {
+	i.s.mu.Lock()
+	defer i.s.mu.Unlock()
+	i.s.counters[arg.Name] += arg.Value
+
+	return true, nil
+}
+
+type FindAllMetricsParams struct {
+}
+
+func (i *Impl) FindAllMetrics(_ context.Context, _ FindAllMetricsParams) ([]Metric, error) {
+	i.s.mu.Lock()
+	defer i.s.mu.Unlock()
+	var metrics []Metric
+	for n, v := range i.s.gauges {
+		metrics = append(metrics, Metric{
+			Type:  enum.GaugeMetricType,
+			Name:  n,
+			Value: v,
+		})
+	}
+	for n, v := range i.s.counters {
+		metrics = append(metrics, Metric{
+			Type:  enum.CounterMetricType,
+			Name:  n,
+			Value: v,
+		})
+	}
+
+	return metrics, nil
+}
+
+type FindCounterMetricParams struct {
+	MetricName string
+}
+
+func (i *Impl) FindCounterMetric(_ context.Context, arg FindCounterMetricParams) (Metric, error) {
+	i.s.mu.Lock()
+	defer i.s.mu.Unlock()
+	metricValue, ok := i.s.counters[arg.MetricName]
+	if !ok {
+		return Metric{}, errorz.ErrUndefinedMetricName
+	}
+
+	return Metric{
+		Type:  enum.CounterMetricType,
+		Name:  arg.MetricName,
+		Value: metricValue,
+	}, nil
+}
+
+type FindGaugeMetricParams struct {
+	MetricName string
+}
+
+func (i *Impl) FindGaugeMetric(_ context.Context, arg FindGaugeMetricParams) (Metric, error) {
```

Comment:
Я бы рекомендовал возвращать указатель на инстанс модели вместо значения. Это позволит избежать лишнего копирования, а также позволит трактовать `nil` более однозначно, как отсутствие значения.
```suggestion
func (i *Impl) FindGaugeMetric(_ context.Context, arg FindGaugeMetricParams) (*Metric, error) {
```